Libraries/src/Amazon.Lambda.RuntimeSupport/Helpers/FileDescriptorLogStream.cs

```diff
@@ -15,17 +15,39 @@ public static class FileDescriptorLogFactory
     {
         private readonly static ConcurrentDictionary<string, StreamWriter> _writers = new ConcurrentDictionary<string, StreamWriter>();
 
+        // max cloudwatch log event size, 256k - 26 bytes of overhead.
+        internal const int MaxCloudWatchLogEventSize = 256 * 1024 - 26;
+        internal const int LambdaTelemetryLogHeaderLength = 8;
+        internal const uint LambdaTelemetryLogHeaderFrameType = 0xa55a0001;
+
         /// <summary>
         /// Get the StreamWriter for the particular file descriptor ID. If the same ID is passed the same StreamWriter instance is returned.
         /// </summary>
         /// <param name="fileDescriptorId"></param>
         /// <returns></returns>
         public static StreamWriter GetWriter(string fileDescriptorId)
+        {
+            SafeFileHandle handle = new SafeFileHandle(new IntPtr(int.Parse(fileDescriptorId)), false);
```

Comment:
Add the creation of the `SafeFileHandle` into Lambda expression for calling `InitializeWriter`. Otherwise we are creating an instance of `SafeFileHandle` every time `GetWriter` is called even if we are returning back a cached `StreamWriter`. This could become problematic given `SafeFileHandle` implements `IDisposable`. 

Something like 

```
            var writer = _writers.GetOrAdd(fileDescriptorId, 
                (x) => {
                    SafeFileHandle handle = new SafeFileHandle(new IntPtr(int.Parse(fileDescriptorId)), false);
                    return InitializeWriter(new FileStream(handle, FileAccess.Write));
                });
            return writer;
```

Reply:
Thanks Norm, fixed!